src/contracts/fund/hub/Hub.ts

```diff
@@ -164,7 +193,25 @@ export class Hub extends Contract {
    * @param from The sender address.
    */
   public setRouting(from: Address) {
-    return this.createTransaction('setRouting', from);
+    const validate = async () => {
+      await this.validateCreator(from);
+
+      let message;
+
+      const isSpokesSet = await this.isSpokesSet();
+      if (!isSpokesSet) {
+        message = `Spokes must be set. isSpokesSet: ${isSpokesSet}`;
```

Comment:
`const message`

Reply:
You don't need to add the `isSpokesSet` parameter to the message.

Reply:
The error should be `SpokesNotSetError` and `RoutesAlreadySetError`. No need for a generic `SetupError`.